### DISCUSSION TITLE
A way to translate validations errors?

### DISCUSSION THREAD

**Discussion by mvrlin** [pocketbase/pocketbase#5401] [Q&A]
Hello! Is there a way to translate validations errors?
1. I found in the docs that you can add a hook before api error, but the error is only about creating a record, not the validation.
2. If not how can I extract the values in length validation for example: `the length must be between 8 and 72`? Looking at the ozzo docs, you can access [variables in string](https://github.com/go-ozzo/ozzo-validation/blob/master/UPGRADE.md)

<img width="960" alt="image" src="https://github.com/user-attachments/assets/eed0b40a-561a-4f22-9f6a-812adeb691d9">

**Comment by ganigeorgiev** [comment]
It is not clear what exactly do you mean by _"translate validations errors"_ (translate how?), nor where and what exactly you are trying to do.

For client-side translations users can use the `code` prop that is returned together with the validation error message (unfortunately the error codes currently are not listed anywhere and may change before stable v1; see also https://github.com/pocketbase/pocketbase/issues/3798).

If you want to access the underlying validation errors you'll have to type assert it to `validation.Errors` or `validation.Error` (in case of an *ApiError, the raw error/data is accessible via [`ApiError.RawData()`](https://pkg.go.dev/github.com/pocketbase/pocketbase@v0.22.20/apis#ApiError.RawData)) but keep in mind that the error could be practically anything and it is not guarantee to have parameters.

_Also note that with the [refactoring](https://github.com/pocketbase/pocketbase/discussions/4355) the `OnBeforeApiError` hook is removed and instead each hook returns its error as part of the `e.Next()` chain call (the same apply for route middlewares since they use the hooks under the hood)._

**Comment by mvrlin** [comment]
@ganigeorgiev Thank you, for your reply. By translating I mean having a dictionary of ozzo errors codes for different languages. I want to pass Accept-Language header and be able to translate API errors accordingly.

**Reply by ganigeorgiev** [reply]
It still not clear to me where do you intend the translation to happen - server-side or client-side. All API errors that returns fields validation has `code` property alongside the `message` (some may also return `params` with the next release) that you can use for client-side validations but as mentioned above we don't have list with all the possible error codes at the moment (_see https://github.com/pocketbase/pocketbase/issues/3798_).

**Comment by lightningspirit** [comment]
I use a [LibreTranslate](https://github.com/LibreTranslate/LibreTranslate) instance to support multi-language in PocketBase. I intercept calls using the provided hooks and translate (effectively caching the results) with LibreTranslate.
